cobalt/black_box_tests/black_box_tests.py

```diff
@@ -243,12 +300,24 @@ def Run(self):
       if self.test_name:
         suite = unittest.TestLoader().loadTestsFromName(_TEST_DIR_PATH +
                                                         self.test_name)
+        return_code = not unittest.TextTestRunner(
+            verbosity=2, stream=sys.stdout).run(suite).wasSuccessful()
+        return return_code
       else:
-        suite = LoadTests(_launcher_params)
-      # Using verbosity=2 to log individual test function names and results.
-      return_code = not unittest.TextTestRunner(
-          verbosity=2, stream=sys.stdout).run(suite).wasSuccessful()
-      return return_code
+        cobalt_tests_return_code = 0
```

Comment:
Do Evergreen tests need to run with a Proxy for any reason ? I'm almost sure they don't - I'd recommend getting it out of this `with Proxy` block - it's notoriously finicky and it's really only needed for _only_ Web Platform tests, even for other Blackboxtests it only adds pain and unreliability and i'm planning to break those out here sometime soon.

Reply:
I tried pulling it out of the `with ProxyServer` block but got this error in the test:
```
[MainWebModule/951761:0519/090427.096454:ERROR:browser_module.cc(1213)] NetFetcher error on http://127.0.0.1:37913/testdata/evergreen_test.html?resetInstallations=true: net::ERR_PROXY_CONNECTION_FAILED, response code -1
```

Can we leave it as is for now and consider removing the Proxy use for Evergreen test cases when it's removed for other tests?

By the way I seem to remember seeing an architecture diagram at some point under `cobalt/black_box_tests` but can't find anything now. I think having something like that would make it easier to contribute to the framework.

Reply:
Sorry i thought i replied here earlier already. If you want to get the code in and running, yes go ahead as is. I'm just expecting you'll be seeing a lot of flakiness trying to run with it, so we'd need to follow up pretty urgently.

The reason why you are getting `ERR_PROXY_CONNECTION_FAILED` is that the command line launching with it is likely still passing the `--proxy` args to Cobalt itself when it runs. I would think that should be not too tricky to prevent.

And yes, there definitely was an ASCII art diagram somewhere -  possibly the one added in go/cobalt-cl/115749 ( there's some scribbled up in http://go/cobalt-testing-pip but that's probably not what you are looking for. It does summarize a few issues with the framework though ). 

tl;dr if this runs as is for you go ahead and submit, don't break your back making it too clean.

Reply:
Ok thanks yeah, I think I'll go ahead and get this in using proxy for now.

And that's the diagram I was thinking of, thanks!